src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1172,6 +1173,54 @@
   (jdbc/do-commands
     "DROP INDEX IF EXISTS resource_events_resource_type_idx"))
 
+(defn merge-fact-values-into-facts
+  []
+  (jdbc/do-commands
+   ["create table facts_transform"
+    "  (factset_id bigint not null,"
+    "   fact_path_id bigint not null,"
+    "   value_integer bigint,"
+    "   value_float double precision,"
+    "   value_type_id integer not null,"
+    "   value_string text,"
+    "   value jsonb,"
+    "   large_value_hash bytea,"
+    "   value_boolean boolean)"]
+
+   ["insert into facts_transform"
+    "  (factset_id,"
+    "   fact_path_id,"
+    "   value_integer,"
+    "   value_float,"
+    "   value_type_id,"
+    "   value_string,"
+    "   value,"
+    "   large_value_hash,"
+    "   value_boolean)"
+    "  select factset_id, fact_path_id,"
+    "         value_integer, value_float, value_type_id, value_string, value,"
+    "         case when pg_column_size(value) >= 40"
+    "              then value_hash"
+    "          end,"
+    "          value_boolean"
+    "    from facts inner join fact_values on fact_value_id = fact_values.id"]
+
+   "drop index facts_fact_value_id_idx"
+   "drop table facts"
+   "drop table fact_values"
+   "drop sequence fact_values_id_seq"
+
+    "alter table facts_transform rename to facts"
+
+    "create index facts_factset_id_idx on facts using btree (factset_id)"
+    "create index facts_fact_path_id_idx on facts using btree (fact_path_id)"
+    "create index facts_value_integer_idx on facts(value_integer)"
+    "create index facts_value_float_idx on facts(value_float)"
```

Comment:
I think we will want indexes for all the columns that previously had them -- this is missing value_string and value_hash